pkg/process/checks/net.go

```diff
@@ -493,11 +493,32 @@ func convertAndEnrichWithServiceCtx(tags []string, tagOffsets []uint32, serviceC
 	return tagsStr
 }
 
-// fetches network_id from the current netNS or from the system probe if necessary, where the root netNS is used
+// retryGetNetworkID attempts to fetch the network_id maxRetries times before failing
 func retryGetNetworkID(sysProbeClient *http.Client) (string, error) {
+	const maxRetries = 3
+	var err error
+	var networkID string
+	for attempt := 1; attempt <= maxRetries; attempt++ {
+		networkID, err = getNetworkID(sysProbeClient)
+		if err == nil {
+			return networkID, nil
+		}
+		log.Debugf(
+			"failed to get network ID from system-probe (attempt %d/%d): %s",
+			attempt,
+			maxRetries,
+			err,
+		)
+		time.Sleep(50 * time.Millisecond)
```

Comment:
maybe do a backoff here? just multiple it by attempt number? 

I dont know if 50ms is correct, seems kinda low but you know better